modules/avro-kafka-exporter/src/test/scala-2.13/io/janstenpickle/trace4cats/avro/kafka/AvroKafkaSpanExporterSpec.scala

```diff
@@ -53,17 +51,15 @@ class AvroKafkaSpanExporterSpec
   it should "serialize a batch of spans to kafka" in withRunningKafkaOnFoundPort(userDefinedConfig) {
     implicit actualConfig =>
       forAll { (batch: Batch[Chunk]) =>
-        AvroKafkaSpanExporter[IO, Chunk](NonEmptyList.one(s"localhost:${actualConfig.kafkaPort}"), "test")
+        val res = AvroKafkaSpanExporter[IO, Chunk](NonEmptyList.one(s"localhost:${actualConfig.kafkaPort}"), "test")
           .use { exporter =>
-            exporter.exportBatch(batch)
+            exporter.exportBatch(batch) >>
+              IO(consumeNumberMessagesFrom[CompletedSpan]("test", batch.spans.size))
           }
           .unsafeRunSync()
 
-        val res = consumeNumberMessagesFrom[CompletedSpan]("test", batch.spans.size)
-
         res.size should be(batch.spans.size)
-        assert(Eq.eqv(res, batch.spans.toList))
+        assert(Eq.eqv(res.sortBy(_.context.spanId.show), batch.spans.toList.sortBy(_.context.spanId.show)))
```

Comment:
It seems that different spans go to different partitions so the order of the whole batch is unstable.